spec/lib/wipe_out/plugins/logger_spec.rb

```diff
@@ -1,59 +1,113 @@
-    logger = double(Logger)
-    logged_messages = []
-    allow(logger).to receive(:debug) { |log| logged_messages << log }
-    user = create(:user, :with_comments)
-    comment = user.comments.first
-    plan.config.logger = logger
-
-    plan.execute(user)
-
-    expect(user.comments).to eq([comment])
-    expect(logged_messages).to match([
-      "[WipeOut] start plan=Plan(attributes=[:last_name, :access_tokens])",
-      "[WipeOut] executing plan=Plan(attributes=[:last_name, :access_tokens]) record_class=User id=#{user.id}",
-      "[WipeOut] executing plan=Plan(attributes=[]) record_class=Comment id=#{comment.id}",
-      "[WipeOut] wiped out plan=Plan(attributes=[]) record_class=Comment id=#{comment.id}",
-      "[WipeOut] wiped out plan=Plan(attributes=[:last_name, :access_tokens]) record_class=User id=#{user.id}",
-      "[WipeOut] completed plan=Plan(attributes=[:last_name, :access_tokens])"
-    ])
+    it "logs all executions for all records, even from relations", :aggregate_failures do
+      logger = double(Logger)
+      logged_messages = []
+      allow(logger).to receive(:debug) { |log| logged_messages << log }
+      user = create(:user, :with_comments)
+      first_comment = user.comments.first
+      second_comment = user.comments.second
+      plan.config.logger = logger
+
+      plan.execute(user)
+
+      expect(user.comments).to eq([first_comment, second_comment])
+      expect(logged_messages).to match([
+        "[WipeOut] start plan=Plan(attributes=[:last_name, :access_tokens])",
+        "[WipeOut] executing plan=Plan(attributes=[:last_name, :access_tokens]) record_class=User id=#{user.id}",
+        "[WipeOut] executing plan=Plan(attributes=[]) record_class=Comment id=#{first_comment.id}",
+        "[WipeOut] wiped out plan=Plan(attributes=[]) record_class=Comment id=#{first_comment.id}",
+        "[WipeOut] executing plan=Plan(attributes=[]) record_class=Comment id=#{second_comment.id}",
+        "[WipeOut] wiped out plan=Plan(attributes=[]) record_class=Comment id=#{second_comment.id}",
+        "[WipeOut] wiped out plan=Plan(attributes=[:last_name, :access_tokens]) record_class=User id=#{user.id}",
+        "[WipeOut] completed plan=Plan(attributes=[:last_name, :access_tokens])"
+      ])
+    end
+
+    describe "nested root plans with plugins" do
+      it "uses plugin only from top level root plan that's executed" do
+        user_plan = WipeOut.build_plan do
+          plugin WipeOut::Plugins::Logger
+          wipe_out :last_name, :access_tokens
+          ignore :comments
+        end
+        user_basic_plan = WipeOut.build_plan do
+          include_plan(user_plan)
+          wipe_out :last_name
+          ignore :comments, :access_tokens
+        end
+        logger = double(Logger)
+        logged_messages = []
+        user_basic_plan.config.logger = logger
+        allow(logger).to receive(:debug) { |log| logged_messages << log }
+
+        plan.execute(create(:user))
```

Comment:
I found that this is a false-positive (or I completely miss the point here) - I think it should be

 ```suggestion
        user_basic_plan.execute(create(:user))
```

to be fixed in another PR.

Reply:
the example(`it`) description is about inheriting from top parent so imho yes, this should be user_basic_plan

Reply:
It didn't fix it here to keep this PR focused and also because after changing to `user_basic_plan` the spec fails :( and it's not related to the changes in this PR. @ktopolski

Reply:
+1, nice spot, should be `user_basic_plan`